### PR TITLE
Feature/transcript

### PR DESCRIPTION
**Pull request by SeniorZhai**

### CODE REVIEW

File: app/src/main/java/one/mixin/android/job/DecryptMessage.kt [hunk 1]
```diff
@@ -617,6 +632,86 @@ class DecryptMessage(private val lifecycleScope: CoroutineScope) : Injector() {
                 messageDao.insertAndNotifyConversation(message, conversationDao, accountId)
                 generateNotification(message, data.source)
             }
+            data.category.endsWith("_TRANSCRIPT") -> {
+                val plain = if (data.category == MessageCategory.PLAIN_TRANSCRIPT.name) String(Base64.decode(plainText)) else plainText
+                val transcripts = gson.fromJson(plain, Array<TranscriptMessage>::class.java).toList()
+                if (transcripts.isEmpty()) {
+                    throw IllegalArgumentException("Empty transcript messages")
+                }
+                val stringBuffer = StringBuffer()
```
Comment (Tougee) [inline app/src/main/java/one/mixin/android/job/DecryptMessage.kt]: perhaps `StringBuilder` would be more efficient

File: app/src/main/java/one/mixin/android/job/SendTranscriptJob.kt [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+        getTranscripts(message.id, transcripts)
+
+        if (transcripts.any { t -> t.isAttachment() }) {
+            val mediaSize = transcripts.sumOf { t -> t.mediaSize ?: 0 }
+            messageDao.updateMediaSize(mediaSize, message.id)
+            transcripts.filter { t ->
+                t.isAttachment()
+            }.forEach { t ->
+                jobManager.addJob(SendTranscriptAttachmentMessageJob(t, message.isPlain(), message.id))
+            }
```
Comment (Tougee) [inline app/src/main/java/one/mixin/android/job/SendTranscriptJob.kt:94]: not handle non-attachment messages?

File: app/src/main/java/one/mixin/android/db/TranscriptMessageDao.kt [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+        t.thumb_image AS thumbImage, t.media_url AS mediaUrl, t.media_width AS mediaWidth, t.media_height AS mediaHeight, st.asset_width AS assetWidth, 
+        st.asset_height AS assetHeight, st.asset_url AS assetUrl, st.asset_type AS assetType,t.media_duration AS mediaDuration, 
+        t.media_waveform AS mediaWaveform, su.user_id AS sharedUserId, su.full_name AS sharedUserFullName, su.avatar_url AS sharedUserAvatarUrl, 
+        su.app_id AS sharedUserAppId, su.identity_number AS sharedUserIdentityNumber, su.is_verified AS sharedUserIsVerified, t.quote_id AS quoteId,
+        t.quote_content AS quoteContent, t.mentions AS mentions
+        FROM transcript_messages t
+        LEFT JOIN users u on t.user_id = u.user_id
+        LEFT JOIN users su ON t.shared_user_id = su.user_id
+        LEFT JOIN stickers st ON st.sticker_id = t.sticker_id
+        WHERE transcript_id = :transcriptId
```
Comment (Tougee) [inline app/src/main/java/one/mixin/android/db/TranscriptMessageDao.kt]: t.transcript_id ?

File: app/src/main/java/one/mixin/android/repository/UserRepository.kt [hunk 4]
```diff
@@ -220,4 +223,6 @@ constructor(
     suspend fun findUserByAppId(appId: String): User? = userDao.findUserByAppId(appId)
 
     fun updateMuteUntil(id: String, muteUntil: String) = userDao.updateMuteUntil(id, muteUntil)
+
+    fun a (transcriptMessages: List<TranscriptMessage>) = transcriptMessageDao.insertList(transcriptMessages)
```
Comment (Tougee) [inline app/src/main/java/one/mixin/android/repository/UserRepository.kt]: should have a new name

File: app/src/main/java/one/mixin/android/job/SendTranscriptJob.kt (hunk 2 above)
Comment (SeniorZhai) [reply app/src/main/java/one/mixin/android/job/SendTranscriptJob.kt:94]: The last step of all attachment upload job is to package the data to send

File: app/src/main/java/one/mixin/android/job/NotificationGenerator.kt [hunk 5]
```diff
@@ -349,12 +365,12 @@ object NotificationGenerator : Injector() {
                                     isFirstResource: Boolean
                                 ):
                                     Boolean {
-                                    notificationBuilder.setLargeIcon(
-                                        BitmapFactory.decodeResource(context.resources, R.drawable.default_avatar)
-                                    )
-                                    notificationManager.notify(message.conversationId.hashCode(), notificationBuilder.build())
-                                    return false
-                                }
+                                        notificationBuilder.setLargeIcon(
```
Comment (crossle) [inline app/src/main/java/one/mixin/android/job/NotificationGenerator.kt]: Follow the  ktlint format

File: app/src/main/java/one/mixin/android/ui/common/UserBottomSheetDialogFragment.kt [hunk 6]
```diff
@@ -496,6 +499,15 @@ class UserBottomSheetDialogFragment : MixinScrollableBottomSheetDialogFragment()
             UserRelationship.FRIEND.name -> {
                 list.groups.add(
                     menuGroup {
+                        // Todo export chat
```
Comment (crossle) [inline app/src/main/java/one/mixin/android/ui/common/UserBottomSheetDialogFragment.kt]: ？

File: app/src/main/java/one/mixin/android/vo/TranscriptMessageItem.kt [hunk 7]
```diff
@@ -0,0 +1,166 @@
...
+    val sharedUserIsVerified: Boolean? = null,
+    val sharedUserAppId: String? = null,
+    val quoteId: String? = null,
+    val quoteContent: String? = null,
+    val mentions: String? = null,
+) : ICategory
+
+fun TranscriptMessageItem.isSignal(): Boolean {
+    return type.startsWith("SIGNAL_")
+}
```
Comment (Tougee) [inline app/src/main/java/one/mixin/android/vo/TranscriptMessageItem.kt]: use `isSignal()` in ICategory

**Review by crossle — APPROVED** [review]

**Pull request merged**